kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java

```diff
@@ -72,44 +77,49 @@ public Optional<GwtNetInterfaceConfig> fillWithStatusProperties(String ifname,
-        if (networkInterface.getUsbDevice() != null) {
-            gwtConfig.setHwUsbDevice(networkInterface.getUsbDevice().getUsbDevicePath());
-        }
+            NetworkInterfaceStatus networkInterfaceStatus) {
+
+        gwtConfig.setHwState(networkInterfaceStatus.getState().name());
+        gwtConfig.setHwType(networkInterfaceStatus.getType().name());
+        gwtConfig.setHwAddress(NetUtil.hardwareAddressToString(networkInterfaceStatus.getHardwareAddress()));
+        gwtConfig.setHwName(networkInterfaceStatus.getName());
+        gwtConfig.setHwDriver(networkInterfaceStatus.getDriver());
+        gwtConfig.setHwDriverVersion(networkInterfaceStatus.getDriverVersion());
+        gwtConfig.setHwFirmware(networkInterfaceStatus.getFirmwareVersion());
+        gwtConfig.setHwMTU(networkInterfaceStatus.getMtu());
+        networkInterfaceStatus.getUsbNetDevice()
+                .ifPresent(usbDevice -> gwtConfig.setHwUsbDevice(usbDevice.getUsbDevicePath()));
+
+        logger.debug("GWT common state properties for interface {}:\\n{}\\n", gwtConfig.getName(),
+                gwtConfig.getProperties());
     }
 
     private void setIpv4DhcpClientProperties(GwtNetInterfaceConfig gwtConfig,
-            NetInterface<NetInterfaceAddress> networkInterface) {
+            NetworkInterfaceStatus networkInterfaceInfo) {
 
         String ipConfigMode = gwtConfig.getConfigMode();
         if (isDhcpClient(ipConfigMode)) {
             /*
              * An interface can have multiple active addresses, we select just the first
              * one. This is a limit of the current GWT UI.
              */
-            if (!networkInterface.getNetInterfaceAddresses().isEmpty()) {
-                NetInterfaceAddress address = networkInterface.getNetInterfaceAddresses().get(0);
-                gwtConfig.setIpAddress(address.getAddress().getHostAddress());
-                gwtConfig.setGateway(address.getGateway().getHostAddress());
-                gwtConfig.setSubnetMask(address.getNetmask().getHostAddress());
-                gwtConfig.setReadOnlyDnsServers(prettyPrintDnsServers(address.getDnsServers()));
-            }
+            networkInterfaceInfo.getInterfaceIp4Addresses().ifPresent(address -> {
+                gwtConfig.setIpAddress(address.getAddresses().get(0).getAddress().getHostAddress());
+                if (address.getGateway().isPresent()) {
+                    gwtConfig.setGateway(address.getGateway().get().getHostAddress());
+                }
+                gwtConfig.setSubnetMask(convertPrefixToNetmask(address.getAddresses().get(0).getPrefix()));
+                gwtConfig.setReadOnlyDnsServers(prettyPrintDnsServers(address.getDnsServerAddresses()));
+            });
         }
     }
 
+    private String convertPrefixToNetmask(short prefix) {
+        SubnetUtils subnetUtils = new SubnetUtils("0.0.0.0" + "/" + prefix);
+        subnetUtils.setInclusiveHostCount(true);
+        return subnetUtils.getInfo().getNetmask();
```

Comment:
The apache stuff might be more robust, but can we consider using this? https://github.com/eclipse/kura/blob/4001a3daf512922ebb486e881eaebca7c350cd67/kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/util/NetworkUtil.java#L44